Margrit/canvasdesign

maccxs opened the pull request:
Im Supermarkt Spiel pausieren wenn Textbox Supermarkt erscheint
    Supermarkt: Du hast Klopapier ergattert - Textbox
    Update des Infziertenstatus nach "GameOver"
    Player darf sich nicht mehr bewegen nach "GameOver"
    Spieler erscheint rechts unten auf dem Canvas etwas größer nach "Game Over"
Fixed bug in supermarket level where the player could hit the same supermarket twice

Commit:
After Game Over infections status update is on

Commit:
The player is placed on the corner of the canvas after failure

* Actually wanted to get coords of canvas corner by script
* This does not work satisfactory, the player is too far to the left
* For the moment the player and friend coords are hardcoded
* Found some more place where we need the levelIsFinished bool was missing

Commit:
Show player also when continue is pressed, fixed placing player in levelmasks

Commit:
Pause game when you have toilet paper

* also gives a message when player has gotten last roll
* removed unsused function in levelmasks for pausing the game
* introduced a comment in levelmasks  since it took me quite long to work out how the game is unpaused

Commit:
Fixed canvas bug in supermarket level

In the supermarket level the player could go to the same sueprmarket
twice to then get the toilet paper. Fixed this by attaching a script
to each supermarket that checks if the player already went there.

Commit:
Forgotten stuff

* scene is now included
* fixed space to be written with capital letter
* changed number of needed masks to  8 again

maccxs committed:
Removed unused bools

holke commented:
Finally had a first glance at this and i like it!
Feels very good and the player on the canvas works perfectly.

Will still need to have  alook at the code.

holke commented:
Minor issue:

In the demo level when i am infected with the virus and conspiracy theories it shows the infected smiley but it doesnt have a sign in its hand.

holke commented on YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs:
```diff
@@ -216,8 +216,8 @@ void Update ()
     {
         if (Time.time - lastTime >= timeBetweenUpdates)
         {
-            Debug.Log("House: inside " + isPlayerInside + " close " + playerIsClose + " crruns " + aCoroutineRuns + " allowed Home: " + endlevel.isPlayerAllowedHome(player)
-                      + " at will: " + EnterAndExitAtWill);
+            //Debug.Log("House: inside " + isPlayerInside + " close " + playerIsClose + " crruns " + aCoroutineRuns + " allowed Home: " + endlevel.isPlayerAllowedHome(player)
+            //          + " at will: " + EnterAndExitAtWill);
             lastTime = Time.time;
         }
```
If you remove the comment you can remove the whole block. Its only purpose was output.
I used it for debugging the house and you are right in that we dont need it anymore.

holke replied:
```suggestion
```

holke commented on YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs:
```diff
@@ -216,8 +216,8 @@ void Update ()
     {
         if (Time.time - lastTime >= timeBetweenUpdates)
         {
```
```suggestion
```

holke commented on YouVsVirus/Assets/Scenes/Levelgethome/YouVsVirus_Levelgethome.unity:
```diff
@@ -868,9 +868,9 @@ Camera:
+    y: 0.00031727552
     width: 1
-    height: 1
+    height: 0.99936545
```
Would love to understand why this happens.

You change  nothing in unity but some magic number in the scene switches from 1 to 0.9993.

maccxs replied:
yes, i find that strange, too.

holke commented on YouVsVirus/Assets/Scenes/levelcollectmasks/SpaceToContinue.cs:
```diff
@@ -6,6 +6,9 @@
 {
+    //FIXME: If this script is always attached to the canvas it handles
+    //we should get the canvas it is attached to in the script and not
+    //assign in the editor
```
This will probably change with the new Menu update anyways.

holke commented on YouVsVirus/Assets/Scenes/leveldemo/EndLevelControllerLeveldemo.cs:
```diff
@@ -19,10 +19,12 @@ protected override void Update()
+            FindAndPlaceHuman(CanvasProp);
+            levelHasFinished = true;
         }
         // make sure that not both canvases appear
```
Why not use else if? This would ensure that only one of the cases can occur.

maccxs replied:
It was no longer needed, so this is done

holke commented on YouVsVirus/Assets/Scenes/leveldisco/EndLevelControllerLeveldisco.cs:
```diff
@@ -16,37 +16,26 @@ public class EndLevelControllerLeveldisco : EndLevelControllerBase
     /// </summary>
     public void PlayerExitHealthyWithoutFriend()
     {
-        CanvasFail.SetActive(true);
-        failText.text = "You got home safely but Sandra is still at the party.";
-    }
-
-    /// <summary>
-    /// Failure due to sick friend or player exit together
-    /// </summary>
-    public void PlayerOrFriendExitSick()
-    {
-        CanvasFail.SetActive(true);
-        failText.text = "Sorry.\nYou got out but were exposed to the virus.";
-    }
-
-    /// <summary>
-    /// Failure due to sick player exit withou friend
-    /// </summary>
-    public void PlayerExitSickWithoutFriend()
-    {
-        CanvasFail.SetActive(true);
-        failText.text = "You were exposed to the virus and Sandra is still at the party.";
```
Why are they gone? Are they elsewhere? They still occur in the game so i assume you moved the code...?

maccxs replied:
They cannot occur anymore. Since the game ends when the player or the friend is infected both cannot move any longer. Hence player and friend cannot exit sick and the player can't exit sick.

holke commented on YouVsVirus/Assets/Scenes/leveldisco/Friend.cs:
```diff
@@ -33,15 +33,15 @@ public override void Start()
             // if the player has not found us
             if (friendFound == false)
             {
                 // we move like the drunken bastard we are
```
:) Lol, nice comment. Never noticed before.

holke commented on YouVsVirus/Assets/Scenes/levelsupermarket/AlreadyWasHere.cs:
```diff
@@ -0,0 +1,23 @@
+/// </summary>
+public class AlreadyWasHere : MonoBehaviour
+{
+   public bool wasHere { get; private set; }
```
```suggestion
    public bool wasHere { get; private set; }
```

holke replied:
Better make this private and add a getter function?

Otherwise i can just change it from the outside and i.e. set it to false again after i was there.

maccxs replied:
I am not sure if I totally understand the concept, but in I think the set function is private like here
https://codeasy.net/lesson/properties
See the example for
 class Student
  {
    public int Age { get; private set; }
  }
?

holke commented on YouVsVirus/Assets/Scenes/levelsupermarket/Supermarket.cs:
```diff
@@ -66,21 +68,29 @@ private IEnumerator PlayerInSupermarket(Player p)
+            CanvasSupermarket.SetActive(true);
+            CanvasSupermarket.GetComponentInChildren<TMP_Text>().text = "You got one last roll of " +
+                                                                         "FeatherSoft Ultra Premium 3D Embossed StrawberryVanilla flavored" +
+                                                                          " toilet paper.\nPress 'Space' to continue.";
```
> FeatherSoft Ultra Premium 3D Embossed StrawberryVanilla flavored

:D :D :D

holke commented on YouVsVirus/Assets/Scripts/Components/EndLevelControllerBase.cs:
```diff
@@ -42,7 +46,7 @@ public abstract class EndLevelControllerBase : MonoBehaviour
     /// already finished.
     /// </summary>
-    protected bool levelHasFinished = false;
+    public bool levelHasFinished = false;
```
Maybe also make this protected again and add a setter/getter function?

Since if the level is finished we dont want anybody to set it to unfinished from the outside.

maccxs replied:
Done with 54a33f274d6085b1f20d557ffda1ed1ef4fdcc26

maccxs replied:
here it is 0b1565d536fbf4aaeee875e6ef1dfad77172deeb

maccxs replied:
Done I think

holke commented on YouVsVirus/Assets/Scripts/Components/EndLevelControllerBase.cs:
```diff
@@ -152,7 +156,80 @@ public virtual void NotifyInt (int data)
+        //    player.transform.position = new Vector3(Mathf.Abs(corners[0].x), -Mathf.Abs(corners[0].y), corners[0].z);
+        //    player.transform.localScale *= 2;
+        //}
+        //else Debug.Log("No child with the name 'Panel' attached to the canvas");
```
Maybe `Panel.GetComponent<Image>().GetWorldCorners` could work?

maccxs replied:
I am pretty sure I tried this first and it gives exactly the same result as   
Panel.GetComponent<RectTransform>().GetWorldCorners(corners);

holke commented on YouVsVirus/Assets/Scripts/levelcollectmasks/ButtonHandleLevelMasks.cs:
```diff
@@ -21,17 +21,4 @@ public void Continue()
-    {
-        CanvaslAllCollectedGetHome.SetActive(false);
-        PauseGame.Unpause();
-    }
```
Where is this now?

maccxs replied:
I think it is now done in SpaceToContinue.cs

holke reviewed (commented):
Very good changes. I have mostly minor comments and questions.

maccxs committed:
Update YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs

Co-authored-by: holke <johannes.holke@dlr.de>

maccxs committed:
Update YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs

Co-authored-by: holke <johannes.holke@dlr.de>

maccxs committed:
Update YouVsVirus/Assets/Scenes/levelsupermarket/AlreadyWasHere.cs

Co-authored-by: holke <johannes.holke@dlr.de>

maccxs commented:
> 
> 
> Minor issue:
> 
> In the demo level when i am infected with the virus and conspiracy theories it shows the infected smiley but it doesnt have a sign in its hand.

That is really strange, can you check again? For me it is working just fine.

holke commented:
![grafik](https://user-images.githubusercontent.com/7125317/84769352-fade7800-afd5-11ea-8088-820f2ba43fc1.png)

It is hard to reproduce since most of the time i get only infected with conspiracy theories.
But after like 10 runs i managed to get it again.

Commit:
protected levelHasFinished

Commit:
Merge branch 'margrit/canvasdesign' of https://github.com/youvsvirus/youvsvirus-unity into margrit/canvasdesign

Commit:
initialized levelhasfinished with false again

Commit:
End level demo

        // the level ends with failure when
        // 1. the player catches the virus only (handled by base class)
        // 2. the player catches the virus and propaganda (below)
        // 3. the player catches proganda only (here)

the notification of the end level controller that the player is infected by proganda
was "too late".  Everything should work properly now.

maccxs commented:
> 
> 
> ![grafik](https://user-images.githubusercontent.com/7125317/84769352-fade7800-afd5-11ea-8088-820f2ba43fc1.png)
> 
> It is hard to reproduce since most of the time i get only infected with conspiracy theories.
> But after like 10 runs i managed to get it again.

Resolved with 
3dda804

maccxs commented:
I think I am finished with your comments. I will research the get/set/init question and comment on that.

Commit:
levelHasFinished as auto-property with initial value

maccxs commented:
The initialization was the other way around as I though it would be and is now included 
df9d705

holke committed:
Merge remote-tracking branch 'origin/master' into margrit/canvasdesign
